topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -104,13 +104,26 @@ public class BatchingTopicController {
         this.selector = Objects.requireNonNull(selector);
         this.useFinalizer = config.useFinalizer();
         this.admin = admin;
-        // Get the config of some broker and check whether auto topic creation is enabled
-        Optional<String> autoCreateValue = getClusterConfig(admin, AUTO_CREATE_TOPICS_ENABLE);
-        if (autoCreateValue.isPresent() ? "true".equals(autoCreateValue.get()) : false) {
+
+        var isCustomSaslConfigSet = config.saslCustomConfigJson() != null && !config.saslCustomConfigJson().isBlank();
```

Comment:
This part can be simplified like this:

```sh
boolean warnOnAutoCreation = config.saslCustomConfigJson() != null && !config.saslCustomConfigJson().isBlank();

if (!warnOnAutoCreation) {
    Optional<String> autoCreateValue = getClusterConfig(admin, AUTO_CREATE_TOPICS_ENABLE); 
    warnOnAutoCreation = autoCreateValue.filter("true"::equals).isPresent();
}

if (warnOnAutoCreation) {
    LOGGER.warnOp("It is recommended that " + AUTO_CREATE_TOPICS_ENABLE + " is set to 'false' " +
        "to avoid races between the operator and Kafka applications auto-creating topics");
}
```

The problem with this is that some managed Kafka services may allow reading cluster config. So using an additional boolean env var could cover both use cases.

Reply:
Ok, that makes sense. Does `ASSUME_AUTO_CREATE_TOPICS` really tell that story though? How about `SKIP_CLUSTER_CONFIG_REVIEW`? This would get used around that topic auto create and also in `warnTooLargeMinIsr()` around `MIN_INSYNC_REPLICAS`.

Reply:
Ok, we now check another cluster configuration when CC integration is enabled, so let's use `STRIMZI_SKIP_CLUSTER_CONFIG_REVIEW`. Please make sure to also add documentation about this.